### PR TITLE
feat: Add localhost:3000/blog page

### PR DESCRIPTION
**Pull request by SAF2k**
This commit introduces a new feature: the addition of a /blog page accessible via localhost:3000. The new page provides a platform for blog content, enhancing the website's functionality. Additionally, it includes the 'github-slugger' package, which aids in generating GitHub-compatible slugs for headings and anchor links, improving documentation and user experience. Further details can be found in the code. Your feedback is welcome as we continue to improve the website's capabilities. #161

### CODE REVIEW

**Comment by SAF2k** [comment]
When hovering over the sidebar menu items, a line appears under the menu items, but this line is crossing the sidebar boundary. The line extends beyond the sidebar's boundary, creating a visual inconsistency. This commit addresses this issue by adjusting the CSS styles responsible for the sidebar menu item hover effect. It ensures that the line stays within the sidebar's confines, providing a cleaner and more visually appealing user interface. Your feedback and suggestions for further improvements are welcome.  #165

**Comment by SAF2k** [comment]
This commit focuses on improving the maintainability and organization of CSS codebase. I've separated one large CSS file into multiple smaller ones to make the code more modular and easier to manage. Each new file is named to reflect its content, ensuring clarity and coherence in the styling approach. This restructuring enhances development workflow and sets the stage for more efficient future updates. Feel free to review the code changes and provide feedback or suggestions.  #144

File: src/components/content/blog/BlogContainer.tsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
+import React from 'react';
```
Comment (anicmarko9) [inline src/components/content/blog/BlogContainer.tsx:1]: I like to separate these import lines with one empty line. That way, I can separate imports coming from 3rd party dependencies, and imports coming from other files, like this:
```diff
import React from 'react';
+                                        
import Tag from './Tag';
import { Post } from '@Mocks/blog-posts';
```

File: src/pages/Blog.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+
+import Layout from '@Layout/Layout';
+import BlogContainer from '@Components/content/blog/BlogContainer';
+import { posts } from '@Mocks/blog-posts';
```
Comment (anicmarko9) [inline src/pages/Blog.tsx:5]: Here you separated nicely.

**Review by anicmarko9 — APPROVED** [review]
I think that it looks pretty good, it's semantically correct. I'll let you know if I find any issues once I put it into production. Thanks for your contribution.

P.S.
- Please run this command before creating PR: `npm run format`

**Pull request merged**